ols/src/llms/providers/openai.py

```diff
@@ -0,0 +1,48 @@
+"""OpenAI provider implementation."""
+
+import logging
+
+from langchain.llms.base import LLM
+from langchain_openai import ChatOpenAI
+
+from ols import constants
+from ols.src.llms.providers import LLMProvider, register_llm_provider_as
+
+logger = logging.getLogger(__name__)
+
+
+@register_llm_provider_as(constants.PROVIDER_OPENAI)
+class OpenAI(LLMProvider):
+    """OpenAI provider."""
+
+    @property
+    def default_params(self):
```

Comment:
It is out of the scope of this PR, but regarding the llm parameters. We can probably use these `default_params` (or rename it to `allowed_params` with defaults) to validate incoming parameters and drop those that are not present in these defaults/allowed.
If we decide the whitelist is how we want to handle this problem.